src/rgw/rgw_auth.cc

```diff
@@ -436,8 +436,12 @@ void rgw::auth::RemoteApplier::load_acct_info(RGWUserInfo& user_info) const
     const rgw_user tenanted_uid(acct_user.id, acct_user.id);
 
     if (rgw_get_user_info_by_uid(store, tenanted_uid, user_info) >= 0) {
-      /* Succeeded. */
-      return;
+      if (user_info.type != info.acct_type) {
```

Comment:
What about the second call to `rgw_get_user_info_by_uid()` that takes the non-tenanted `uid` few lines below?

Additionally, maybe the check could be placed in a dedicated, private method.